### PR TITLE
feat: add teleportr API server

### PR DESCRIPTION
**Pull request by cfromknecht**
**Description**
This PR adds the teleport API server that will be used to facilitate requests for the Gateway and track deposits/disbursements. Posting this sketch to get early feedback on the desired API.

**Metadata**
- Fixes ENG-1933

### CODE REVIEW

File: go/teleportr/api/server.go [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+		return
+	}
+	address := common.HexToAddress(addresses[0])
+
+	ctx, cancel := context.WithTimeout(s.ctx, defaultTimeout)
+	defer cancel()
+
+	gasTipCap, err := s.l1Client.SuggestGasTipCap(ctx)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
```
Comment (mslipper) [inline go/teleportr/api/server.go:230]: We should log all these errors - right now they'll be swallowed.

Comment (mslipper) [reply go/teleportr/api/server.go:230]: I found that wrapping the handlers in an `ErrorTrackingHandler` that logs errors and writes `http.StatusInternalServerError` is a clean way to do it. This also provides a hook for you to capture error metrics as well.

**Review by mslipper — CHANGES_REQUESTED** [review]
Should also add some HTTP request metrics as well - e.g., # of requests per path, # of HTTP errors, etc.

File: go/teleportr/api/server.go (hunk 1 above)
Comment (cfromknecht) [reply go/teleportr/api/server.go]: added! thanks for the suggestion 👍

**Review by mslipper — APPROVED** [review]

**Pull request merged**